clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1817,8 +1831,16 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
     // sampler __init method has only one argument
     const ParmVarDecl *SamplerArg = InitMethod->getParamDecl(0);
     assert(SamplerArg && "sampler __init method must have sampler parameter");
+    uint64_t Size;
+    QualType ArgTy = SamplerArg->getType();
+    const ConstantArrayType *CAT =
+        SemaRef.getASTContext().getAsConstantArrayType(ArgTy);
+    if (CAT)
+      ArgTy = CAT->getElementType();
+    Size = SemaRef.getASTContext().getTypeSizeInChars(ArgTy).getQuantity();
 
-    addParam(FD, SamplerArg->getType(), SYCLIntegrationHeader::kind_sampler);
+    Header.addParamDesc(SYCLIntegrationHeader::kind_sampler, static_cast<unsigned>(Size),
```

Comment:
I replaced `addParam` with `addParamDesc` and added code from [here](https://github.com/srividya-sundaram/llvm/pull/5/files#diff-c88a590aca6d5f8e92dd1f8b6ec05a49R1766)

Reply:
See my comment above.